pkg/events/correlation_context.go

```diff
@@ -25,6 +25,7 @@ import (
 type correlationKey struct{}
 
 // ContextWithCorrelationID returns a derived context with the correlation IDs if they were not already in there.
+// NOTE: This function assumes that the provided correlation IDs do not contain any duplicates.
```

Comment:
It's not very expensive to check for + remove duplicates, especially since we're already sorting the thing. Maybe consider doing that instead of making the caller responsible for it.

Reply:
e84bcc956c18f783bb3bd235c6adc604c29cdea5